### PR TITLE
Add env option to build

### PR DESCRIPTION
**Pull request by afinch7**
This should allow us to fix the current issue with windows builds in deno.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -79,6 +81,9 @@ pub fn build(target: &str) {
   cmd.arg("-C");
   cmd.arg(&gn_out_dir);
   cmd.arg(target);
+  for item in env {
+    cmd.env(item.0, item.1);
+  }
```
Comment (ry) [inline src/lib.rs]: Note that ninja is also called in src/deps.rs ... Probably you should change the `ninja()` function to take an env and gn_out_dir and return a `Command`

**Review by ry — COMMENTED** [review]
Cool - what are the env items that need to be passed for windows?

Looks like you need to update the example to get CI to pass.

FYI I have this little script for testing locally
```bash
#!/bin/bash
export NINJA=/Users/rld/src/deno/third_party/depot_tools/ninja
export GN=/Users/rld/src/deno/buildtools/mac/gn
source test.sh
```

**Comment by afinch7** [comment]
> Cool - what are the env items that need to be passed for windows?

Mainly python stuff. See the old code:
https://github.com/denoland/deno/blob/2e1ab8232156a23afd22834c1e707fb3403c0db6/tools/gn.rs#L89-L115

Trying this out in denoland/deno#2698
Working on a cleanup/fixes patch for this PR while I wait.

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**